src/main/webapp/javascript/user-page.js

```diff
@@ -73,33 +138,27 @@ setUpDivWithNoRecipes = (divID, message) => {
   recipesDiv.style.height = "100px";
 }
 
-// opens the recipe form modal
-function openModal() {
-  document.getElementById("recipe-modal").style.display = "block";
-  fetchBlobstoreUrl();
-}
-
 // not quite sure what this does; something with blobs
-function fetchBlobstoreUrl() {
-  fetch('/blobstore-upload-url')
+function fetchBlobstoreUrl(id) {
+  fetch('/blobstore-upload-url?divID='+id)
       .then((response) => {
         return response.text();
       })
       .then((imageUploadUrl) => {
-        const messageForm = document.getElementById('recipe-form');
-        messageForm.action = imageUploadUrl;
+        let form;
+        if (id == "new-recipe-modal") {
+          form = document.getElementById("new-recipe-form");
+        } else if (id == "edit-recipe-modal") {
```

Comment:
This is just a general comment, but I've noticed that a lot of your equality operators use ==, and I know that JS has different behavior for === and ==. If you're aware of that and are sure about your decisions, then that's fine, but I wanted to point it out because it can cause some pretty weird situations.